### PR TITLE
Newlib enhancements

### PR DESCRIPTION
**Pull request by deece**
This pull request adds a number of Newlib options, and adds a new sample configuration (arm-nano-eabi), intended to allow the smallest possible binaries to be produced.

The config argument generation has been refactored from a number of if blocks to a list and a loop that traverses it.

This obsoletes #371

### CODE REVIEW

File: scripts/build/libc/newlib.sh [hunk 1]
```diff
@@ -87,17 +77,44 @@ do_libc() {
...
+REENT_SMALL:reent-small
+MULTITHREAD:multithread
+WIDE_ORIENT:newlib-wide-orient
+UNBUF_STREAM_OPT:unbuf-stream-opt
+ENABLE_TARGET_OPTSPACE:target-optspace
+    "
+
+    for ynarg in $yn_args; do
+        var="CT_LIBC_NEWLIB_${ynarg%:*}"
+	eval var=\$${var}
```
Comment (stilor) [inline scripts/build/libc/newlib.sh]: Formatting glitch (tab vs space)

File: scripts/build/libc/newlib.sh [hunk 2]
```diff
@@ -125,16 +128,18 @@ ENABLE_TARGET_OPTSPACE:target-optspace
...
-        --target=${CT_TARGET}                           \
-        --prefix=${CT_PREFIX_DIR}                       \
-        "${newlib_opts[@]}"                             \
+    CT_DoExecLog CFG                                               \
+    CC_FOR_BUILD="${CT_BUILD}-gcc"                                 \
+    CFLAGS_FOR_TARGET="${cflags_for_target}"                       \
+    AR_FOR_TARGET="`which ${CT_TARGET}-gcc-ar`"                    \
+    RANLIB_FOR_TARGET="`which ${CT_TARGET}-gcc-ranlib`"            \
+    ac_cv_path_AR_FOR_TARGET="`which ${CT_TARGET}-gcc-ar`"         \
+    ac_cv_path_RANLIB_FOR_TARGET="`which ${CT_TARGET}-gcc-ranlib`" \
```
Comment (stilor) [inline scripts/build/libc/newlib.sh]: Can you explain why both `AR_FOR_TARGET` and `ac_cv_path_AR_FOR_TARGET` are needed? Is it for different versions of newlib, or does newlib indeed require both?

In any case, a comment with an explanation is warranted.

File: scripts/build/libc/newlib.sh (hunk 1 above)
Comment (deece) [inline scripts/build/libc/newlib.sh]: Ok

File: scripts/build/libc/newlib.sh (hunk 2 above)
Comment (deece) [inline scripts/build/libc/newlib.sh]: I just tested without ac_cv_path_AR/RANLIB_FOR_TARGET and it builds fine. These were introduced when I was having trouble making the build use the update ar/ranlib, but that was a workflow problem (I didn't realise the scripts were copied sideways). I'll remove them.

**Comment by stilor** [comment]
I missed this in the review, but this new sample actually uses 'unknown' as the vendor part of the tuple, not 'nano'. As a result, 'ct-ng saveconfig' will save this configuration into a wrong directory and the installation path will not be matching the sample name.

Please fix and create a new PR - I'll approve & merge.

**Pull request merged**